src/graphnet/data/curated_datamodule.py

```diff
@@ -271,12 +280,282 @@ def prepare_data(self) -> None:
         """Prepare the dataset for training."""
         assert self._file_hashes is not None  # mypy
         file_hash = self._file_hashes[self._backend]
+        file_path = os.path.join(self.dataset_dir, file_hash + ".tar.gz")
         if os.path.exists(self.dataset_dir):
             return
         else:
             # Download, unzip and delete zipped file
+
+            os.makedirs(self.dataset_dir, exist_ok=True)
+
             os.makedirs(self.dataset_dir)
-            file_path = os.path.join(self.dataset_dir, file_hash + ".tar.gz")
+            _, file_name = os.path.split(file_hash)
+            extension = ".tar.gz" if ".tar.gz" not in file_name else ""
+            file_path = os.path.join(
+                self.dataset_dir,
+                file_name + extension,
+            )
+
             os.system(f"wget -O {file_path} {self._mirror}/{file_hash}")
-            os.system(f"tar -xf {file_path} -C {self.dataset_dir}")
+            print("Unzipping file, this might take a while..")
+            if self._backend == "parquet":
+                os.system(f"tar -xf {file_path} -C {self.dataset_dir}")
+            else:
+                os.system(f"tar -xvzf {file_path} -C {self.dataset_dir}")
             os.system(f"rm {file_path}")
+
+
+class PublicBenchmarkDataset(ERDAHostedDataset):
+    """A generic class for public Prometheus Datasets hosted using ERDA."""
+
+    def __init__(
+        self,
+        graph_definition: GraphDefinition,
+        download_dir: str,
+        backend: str = "parquet",
+        mode: str = "train",
```

Comment:
I do not quite understand why the naming of the different modes are "train/test/test-no-noise".